pkg/suggestion/nasrl_service.py

```diff
@@ -1,22 +1,227 @@
+                  "train_op": self.controller.train_op,
+                  "lr": self.controller.lr,
+                  "grad_norm": self.controller.grad_norm,
+                  "optimizer": self.controller.optimizer,
+                  "baseline": self.controller.baseline,
+                  "entropy": self.controller.sample_entropy,
+                  "sample_arc": self.controller.sample_arc,
+                  "skip_rate": self.controller.skip_rate}
+
+            run_ops = [
+                controller_ops["loss"],
+                controller_ops["entropy"],
+                controller_ops["lr"],
+                controller_ops["grad_norm"],
+                controller_ops["baseline"],
+                controller_ops["skip_rate"],
+                controller_ops["train_op"]]
+
+            if self.is_first_run:
+                print("First time running suggestion. Random architecture will be given. \n")
+                with tf.Session() as sess:
+                    sess.run(tf.global_variables_initializer())
+                    arc = sess.run(controller_ops["sample_arc"])
+                    saver.save(sess, self.ctrl_cache_file)
+
+                self.is_first_run = False
+
+            else:
+                with tf.Session() as sess:
+                    saver.restore(sess, self.ctrl_cache_file)
+
+                    valid_acc = self.controller.reward
+                    result = self.GetEvaluationResult(request.study_id)
+                    loss, entropy, lr, gn, bl, skip, _ = sess.run(
+                        fetches=run_ops,
+                        feed_dict={valid_acc: result})
+                    print("Suggetion updated. LSTM Cell Loss:", loss)
+                    arc = sess.run(controller_ops["sample_arc"])
+
+                    saver.save(sess, self.ctrl_cache_file)
+
+        arc = arc.tolist()
+        organized_arc = [0 for _ in range(self.num_layers)]
+        record = 0
+        for l in range(self.num_layers):
+            organized_arc[l] = arc[record: record + l + 1]
+            record += l + 1
+
+        nn_config = dict()
+        nn_config['num_layers'] = self.num_layers
+        nn_config['input_size'] = self.input_size
+        nn_config['output_size'] = self.output_size
+        nn_config['embedding'] = dict()
+        for l in range(self.num_layers):
+            opt = organized_arc[l][0]
+            nn_config['embedding'][opt] = self.search_space[opt].get_dict()
+
+        organized_arc_json = json.dumps(organized_arc)
+        nn_config_json = json.dumps(nn_config)
+
+        organized_arc_str = str(organized_arc_json).replace('\"', '\'')
+        nn_config_str = str(nn_config_json).replace('\"', '\'')
+
+        print("\nNew Neural Network Architecture (internal representation):")
+        print(organized_arc_json)
+        print("\nCorresponding Seach Space Description:")
+        print(nn_config_str)
+        print()
+
         trials = []
+        trials.append(api_pb2.Trial(
```

Comment:
always only return one trial, ignoring requestNumber?

Reply:
At the current stage, we will return one trial each time. We will add support for multiple trials in the near future.

Reply:
OK, maybe you can create an issue to trace it

Reply:
Sure, I will add a TODO here.